### ISSUE TITLE
continuity/prospector_beta: sfh bin_edges_gyr is accepted but never reaches the forward pass

### ISSUE THREAD

**Issue by suchethac** [suchethac/tengri#1975]
Follow-up to #337, which is closed but whose second acceptance criterion is not met: `sfh={'bin_edges_gyr': ...}` reaches `model.spec` and goes no further. The forward pass uses the default 13.7 Gyr ladder whatever you pass, and nothing warns.

## Reproducer

Two models differing only in `bin_edges_gyr`, at z = 1.5:

```python
EDGES = np.concatenate([[0.0, 0.03], np.logspace(np.log10(0.1), np.log10(float(age_at_z(1.5))), 6)])
# [0, 0.03, 0.1, 0.212, 0.449, 0.953, 2.020, 4.283]

m_def = SEDModel.build(..., sfh={"type": "continuity", "all_params": FREE})
m_new = SEDModel.build(..., sfh={"type": "continuity", "all_params": FREE, "bin_edges_gyr": EDGES})
```

With the same parameter dict (all six ratios 0.4):

```
 default: max lbt with SFR>0 = 13.804 Gyr  SFR(0.01Gyr)=81.56  SFR(3Gyr)=1.842  SFR(8Gyr)=0.3247
          phot[0:3]=[4.34295978e-29 9.66879296e-29 9.49107134e-29]
  custom: max lbt with SFR>0 = 13.804 Gyr  SFR(0.01Gyr)=81.56  SFR(3Gyr)=1.842  SFR(8Gyr)=0.3247
          phot[0:3]=[4.34295978e-29 9.66879296e-29 9.49107134e-29]
```

Bit-identical photometry is the signature of a config that never reaches the graph.

The registry layer is fine. `resolve_sfh("continuity", bin_edges_gyr=EDGES)` returns a genuinely different SFH from `resolve_sfh("continuity")`, and the raw `continuity(...)` honors the kwarg too. `model.spec.bin_edges_gyr` also holds the array. The value is lost between the spec and the forward pass: `SEDModel._init_sfh` (`forward/sed_model.py:2358`) reads `getattr(spec, "bin_edges_gyr", None)` and forwards it, but the live build path is the stellar `SEDModelComponent`, whose `apply` (`components/stellar/component.py:2173`) resolves `SFH_REGISTRY[self.config.sfh_model]` and builds `sfh_kwargs` from `internal_param_map` alone. The config carries no bin edges, so the component never sees them.

## Why the guard did not catch it

`tests/contract/test_bin_edges_gyr_plumbing.py::TestForwardPassPropagation` is named for the forward pass and documented as "`SEDModel.build` actually wires the edges into the SFH function", but `test_build_uses_supplied_edges` asserts only that the array is on the spec, and then records in a comment that "no exception is the contract". Nothing evaluates the model. `TestOtherNonparametricSfhs` likewise stops at `parse_groups`. The suite is green and the user-facing path has never worked.

## Why it matters

The bin ladder is fixed to 13.7 Gyr while the CSP kernel zeroes star formation older than the universe at the fit redshift (`sfr_valid = jnp.where(age < t_obs_gyr * 1e9, sfr_ext, 0.0)`, `components/stellar/component.py:895`). At z = 1.5, where the universe is 4.28 Gyr old, the 3-6 and 6-13.7 Gyr bins sit entirely outside the observable range. Two of the six ratios therefore take no likelihood and sample their Student-t prior, which costs sampler time and widens the posterior for nothing. `sfh_cont_log_total_mass` normalizes over the full ladder (`mass_unnorm = jnp.sum(sfr_unnorm * bin_widths_yr)`, `sfh/nonparametric.py:108`) including the masked bins, so it is not the mass the model actually forms; for a flat SFH at z = 1.5 roughly two thirds of the nominal mass falls in masked bins. Redshift-matched edges are the documented fix and are exactly what cannot currently be set.

This is the whole point of `prospector_beta`, per #337, and it blocks any Prospector-comparable nonparametric fit at z > 0.

## Acceptance criteria

- [ ] Two models differing only in `bin_edges_gyr` produce different `predict_photometry` output for the same parameters.
- [ ] The stellar component config carries the edges and `apply` uses them, so the modern build path and `SEDModel._init_sfh` agree.
- [ ] Rewrite `TestForwardPassPropagation` to assert on model output, not on spec attributes. The bin layout is checkable: the SFR step boundaries in `state.derived["sfr_history"]` must fall at the supplied edges.
- [ ] Either honor the kwarg or reject it loudly; silently accepting a dropped config is the failure mode.
- [ ] Consider warning when a nonparametric bin edge exceeds the age of the universe at the fit redshift, since that bin is unconstrained by construction.

Found while writing `notebooks/jwst_nonparametric_fits.py`; that page documents the current behavior rather than teaching the broken kwarg.

**Comment by suchethac** [comment]
Fixed on main by 169471340 ("fix(sfh,forward): sfh bin_edges_gyr never reached the forward pass"): the stellar component now consumes the resolve_sfh partial (so the edges reach the forward pass), and compile_signature includes the bin layout (so two models differing only in edges no longer share a compiled kernel).

Verified on current main: `tests/regression/bug/test_bug_1975_bin_edges_forward_pass.py` — 4 passed. The test asserts on model output rather than spec attributes, which is the gap that let #337 close while broken.
